ETLParser.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python
+
+# Parser for PEDSnet ETL conventions document
+
+import json
+import re
+import string
+import sys
+
+_field = re.compile(r'^(?P<name>[^\|]*)\|(?P<req>[^\|]*)'
+                    r'\|(?P<type>[^\|]*)\|(?P<desc>[^\|]*)'
+                    r'(\|(?P<conv>[^\|]*))?\|?$')
+_h1 = re.compile(r'^\#[^\#]+')
+_h2 = re.compile(r'^\#\#[^\#]+')
+_table_header = re.compile(r'^Field\s*\|\s*Required\s*\|')
+_table_name = re.compile(r'^\#\#\s*\d+\.\d+\s*(?P<name>.*)$')
+
+
+def _get_table_name(l):
+    return _table_name.match(l).group('name').lower()
+
+
+def _get_field(l):
+    '''
+    Read field information. The format is
+    'Field |Required | Data Type | Description | PEDSnet Conventions',
+    where the 'PEDSnet Conventions' field is optional.
+    '''
+    field = _field.match(l)
+
+    if not field:
+        return None
+
+    name = field.group('name').lower().strip()
+    req = field.group('req').lower().strip().startswith('yes')
+    data_type = field.group('type').strip()
+    desc = field.group('desc').strip()
+    conv = field.group('conv')
+    if conv:
+        conv = conv.strip()
+    else:
+        conv = ''
+
+    return (name, req, data_type, desc, conv)
+
+
+class Document():
+    def __init__(self, fileobj):
+        self.dict = {}
+        self.dict['content'] = ''
+        self.dict['tables'] = {}
```

Comment:
Clean up all of the dict initializations to reduce the number of lookups in the dictionary itself and make it more readable.

```python
self.model = {
    'content': '',
    'tables': {},
}
```